vitrina/datasets/models.py

```diff
@@ -467,6 +477,49 @@ def get_download_count(self):
             )
         )["model_requests__sum"] or 0
 
+    def save_translation(self, translation, *args, **kwargs):
+        if translation.language_code == 'lt':
+            if not self.has_translation(language_code='en'):
+                lt_title = self.lt_title()
+                lt_description = self.lt_description()
+
+                self.create_translation(language_code='en')
+                self.set_current_language('en')
+
+                response_title = requests.post(
+                    "https://vertimas.vu.lt/ws/service.svc/json/Translate",
+                    json={
+                        "appId": "",
+                        "systemID": "smt-8abc06a7-09dc-405c-bd29-580edc74eb05",
+                        "text": lt_title,
+                        "options": ""
+                    },
+                    headers={
+                        "client-id": TRANSLATION_CLIENT_ID,
+                        "Content-Type": "application/json; charset=utf-8"
+                    },
+                )
```

Comment:
It would be nice, if translation would be disabled if TRA.SLATION_CLIENT_ID is not given.

Also, it would be nice if form data saving would work even if VU service is down or misbehaving, for that we need to set [timeout to max 1 second](https://docs.python-requests.org/en/latest/user/advanced/#timeouts) and [handle possible request errors](https://requests.readthedocs.io/en/latest/user/quickstart/#errors-and-exceptions), probably we should handle all `RequestException` errors, and fail silently, adding just a log message about the error.

Also, since here we do two requests, maybe it would be better to do in [under single session](https://docs.python-requests.org/en/latest/user/advanced/#session-objects), to make use of Keep Alive and improve performance and cause less stress on VU servers?

Reply:
Created a separate task for this:

- https://github.com/atviriduomenys/katalogas/issues/770